src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/DataAnnotationsModelMetadataDetailsProvider.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Metadata
+{
+    public class DataAnnotationsModelMetadataDetailsProvider : 
+        IModelMetadataBindingDetailsProvider,
+        IModelMetadataDisplayDetailsProvider
+    {
+        public void GetBindingDetails([NotNull] ModelMetadataBindingDetailsContext context)
+        {
+            context.BindingDetails.IsRequired = context.Attributes.OfType<RequiredAttribute>().Any();
+
+            var editableAttribute = context.Attributes.OfType<EditableAttribute>().FirstOrDefault();
+            if (editableAttribute != null)
+            {
+                context.BindingDetails.IsReadOnly = !editableAttribute.AllowEdit;
+            }
+        }
+
+        public void GetDisplayDetails([NotNull] ModelMetadataDisplayDetailsContext context)
+        {
+            SetDisplayDetails(context.Attributes, context.DisplayDetails);
+        }
+
+        private static void SetDisplayDetails(IReadOnlyList<object> attributes, ModelMetadataDisplayDetails details)
```

Comment:
I'm super-duper unconcerned about this - the caching story is far far improved already compared to what we had before. Let's profile it and see where improvements can be made in the future.